### DISCUSSION TITLE
Rb_gc_register_address usage

### DISCUSSION THREAD

**Discussion by aardvark179** [ohler55/oj#815] [General]
Is there a reason for registering all the constants defined under the `Oj` module explicitly with the go? It appears to have been done earlier this year, but I don’t see a corresponding bug.

it shouldn’t be necessary as those constants are references from the module, and so will live as long as it does.

just trying to understand why they were added rather than simply removing them while refactoring the marking code.

**Comment by ohler55** [comment]
With compaction the values may move so the constants would no longer point to the correct value. Compaction is relatively new for Ruby so it was not needed before that feature was added.

**Reply by aardvark179** [reply]
Ah, and because you are referring to them by the static variable this could be an issue.

the Ruby C API is hell.

**Reply by ohler55** [reply]
It certainly takes a bit of knowledge to comply with and without much documentation on all the API calls it can get tough to work with.
